pkg/controller/datavolume-controller.go

```diff
@@ -455,25 +459,29 @@ func (r *DatavolumeReconciler) Reconcile(_ context.Context, req reconcile.Reques
 		}
 	}
 
-	if selectedCloneStrategy == SmartClone || selectedCloneStrategy == CsiClone {
-		if selectedCloneStrategy == CsiClone {
-			if pvc.Status.Phase == corev1.ClaimBound {
-				if err := r.setCloneOfOnPvc(pvc); err != nil {
-					return reconcile.Result{}, err
-				}
-			} else if pvc.Status.Phase == corev1.ClaimPending {
-				return reconcile.Result{}, r.updateCloneStatusPhase(cdiv1.CSICloneInProgress, datavolume, pvc, selectedCloneStrategy)
-			} else if pvc.Status.Phase == corev1.ClaimLost {
-				return reconcile.Result{},
-					r.updateDataVolumeStatusPhaseWithEvent(cdiv1.Failed, datavolume, pvc, selectedCloneStrategy,
-						DataVolumeEvent{
-							eventType: corev1.EventTypeWarning,
-							reason:    ErrClaimLost,
-							message:   fmt.Sprintf(MessageErrClaimLost, pvc.Name),
-						})
-			}
+	switch selectedCloneStrategy {
+	case HostAssistedClone:
+		if err := r.ensureExtendedToken(pvc); err != nil {
+			return reconcile.Result{}, err
 		}
-
+	case CsiClone:
+		if pvc.Status.Phase == corev1.ClaimBound {
+			if err := r.setCloneOfOnPvc(pvc); err != nil {
+				return reconcile.Result{}, err
+			}
+		} else if pvc.Status.Phase == corev1.ClaimPending {
+			return reconcile.Result{}, r.updateCloneStatusPhase(cdiv1.CSICloneInProgress, datavolume, pvc, selectedCloneStrategy)
+		} else if pvc.Status.Phase == corev1.ClaimLost {
+			return reconcile.Result{},
+				r.updateDataVolumeStatusPhaseWithEvent(cdiv1.Failed, datavolume, pvc, selectedCloneStrategy,
+					DataVolumeEvent{
+						eventType: corev1.EventTypeWarning,
+						reason:    ErrClaimLost,
+						message:   fmt.Sprintf(MessageErrClaimLost, pvc.Name),
+					})
+		}
+		fallthrough
```

Comment:
explicit fallthrough, nice, 
looks better than previous if else is

Reply:
maybe should change embedded `if/else if` to `switch pvc.Status.Phase` as well for consistency